tests/src/core/testqgsfeaturerequest.cpp

```diff
@@ -0,0 +1,153 @@
+/***************************************************************************
+     testqgsfeaturerequest.cpp
+     -------------------------
+    Date                 : May 2021
+    Copyright            : (C) 2021 Sandro Santilli
+    Email                : strk at kbt dot io
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgstest.h"
+#include <QObject>
+#include <QString>
+#include <QStringList>
+#include <QSettings>
+
+#include "qgsfeature.h"
+#include "qgsfield.h"
+#include "qgsgeometry.h"
+#include "qgssymbol.h"
+#include "qgssimplifymethod.h"
+#include "qgslinesymbol.h"
+#include "qgsexpressioncontextutils.h"
+
+class TestQgsFeatureRequest: public QObject
+{
+    Q_OBJECT
+
+  private:
+    void testDefaultConstructed( const QgsFeatureRequest &f );
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void init();// will be called before each testfunction is executed.
+    void cleanup();// will be called after every testfunction.
+    void constructorTest(); //test default constructors
+    void copyConstructorTest(); //test copy constructor
+    void assignmentOperatorTest(); //test copy constructor
+
+
+  private:
+
+};
+
+void TestQgsFeatureRequest::initTestCase()
+{
+
+}
+
+void TestQgsFeatureRequest::cleanupTestCase()
+{
+
+}
+
+void TestQgsFeatureRequest::init()
+{
+
+}
+
+void TestQgsFeatureRequest::cleanup()
+{
+
+}
+
+void TestQgsFeatureRequest::testDefaultConstructed( const QgsFeatureRequest &f )
```

Comment:
Note that there's already a bunch of tests in test_qgsfeaturerequest.py -- there may be duplicates here now

Reply:
It's not a blocker, but in general it's nicer to keep the tests together so that it avoids duplication of effort